api/swapi.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+const got = require('got');
+
+const paths = {
+    people: 'people/',
+    films: 'films/',
+    species: 'species/',
+    planet: 'planets/'
+}
+
+
+class Swapi {
+
+    constructor(path) {
+        this.baseUrl = 'https://swapi.co/api/';
+        this.objectId = path;
+        this.gotDefaultOption = {
+            json: true
+        };
+    }
+
+    async get(url, options = this.gotDefaultOption) {
+        return got(url, options);
+    }
+
+    async getDetailedCharacter(characterId) {
+        const url = `${this.baseUrl}${paths.people}${characterId}`;
+        
+        try {
+            const character = await this.get(url);
+            const species = await this.get(character.body.species[0]);
+            const homeWorld = await this.get(character.body.homeworld);
+            const films = await Promise.all(character.body.films.map(film=>this.get(film)));
+            return { ...character.body, homeworld: homeWorld.body, species: species.body, films: films.map((film)=>film.body) };
+        } catch (e) {
+            console.log(e)
```

Comment:
```suggestion
            console.log(e);
            throw e;
```

This kicks the error up the chain. It's good to pass the error along so the caller can  handle errors the way they need to. It's not up to the Swapi class to decide that